documentation/modules/oauth/con-oauth-reauthentication.adoc

```diff
@@ -1,23 +1,41 @@
+OAuth 2.0 authentication in Strimzi supports _session re-authentication_, which is part of the Kafka Client libraries. 
+This Kafka mechanism is used to enforce the expiry of authenticated sessions between OAuth clients and Kafka brokers after a defined period of time. 
+When a session expires, the client immediately starts a new session by reusing the existing connection rather than dropping it.
 
-When the re-authentication mechanism is enabled in the configuration of an `oauth` type listener, the broker's authenticated session expires when the access token expires.
-The client must then re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
+Session re-authentication is disabled by default. 
+To enable and configure it, you set a time value for `maxSecondsWithoutReauthentication` in the `oauth` listener configuration in the `Kafka` custom resource. 
+The same property is used to configure session re-authentication for OAUTHBEARER and PLAIN authentication. 
+For an example configuration, see xref:proc-oauth-authentication-broker-config-{context}[].
+
+Session re-authentication can be used with _fast local JWT_ or _introspection endpoint_ token validation.
+
+.Client re-authentication
+
+When the broker's authenticated session expires, the client must re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
 
 If token validation is successful, a new client session is started using the existing connection.
 If the client fails to re-authenticate, the broker will close the connection if further attempts are made to send or receive messages.
 Java clients that use Kafka client library 2.2 or later automatically re-authenticate if the re-authentication mechanism is enabled on the broker.
 
-You enable session re-authentication in the `Kafka` resource. Set the `maxSecondsWithoutReauthentication` property for a TLS listener with `type: oauth` authentication.
-Session re-authentication is supported for both types of token validation (fast local JWT and introspection endpoint). For an example configuration, see xref:proc-oauth-authentication-broker-config-{context}[].
+.Differences in session expiry for OAUTHBEARER and PLAIN
+
+There are some differences in how session expiry works when session re-authentication is configured for OAUTHBEARER and PLAIN authentication.
 
-For more information about the re-authentication mechanism, which was added in Kafka version 2.2, see link:https://cwiki.apache.org/confluence/display/KAFKA/KIP-368%3A+Allow+SASL+Connections+to+Periodically+Re-Authenticate[KIP-368^].
+* For OAUTHBEARER, the broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
+The session will expire earlier if the access token expires before the `maxSecondsWithoutReauthentication`. 
```

Comment:
What about refresh token?

Reply:
Refresh token is never sent to the Kafka broker for authentication. It is only saved in Kafka client configuration, when configured, and is used by client to obtain a fresh new access token as needed, which is then sent to the Kafka broker.

Reply:
@mstruk - Does session re-auth (with OAUTHBEARER) apply to refresh tokens?

Reply:
Re-authentication is configured and triggered by Kafka broker so it applies to Kafka clients regardless of how they are configured. The refresh token is on the Kafka client. Re-authentication applies. The client refreshes the access token by using the refresh token, and uses the new access token to re-authenticate.

Reply:
Thanks, well spotted. I added a new part about refresh tokens.

> Session re-authentication also applies to refresh tokens, if used. When the session expires, the client refreshes the access token by using its refresh token. The client then uses the new access token to re-authenticate to the existing session.